src/app.ts

```diff
@@ -31,14 +32,15 @@ app.use((_req: Request, res: Response, next: NextFunction): void => {
 if (process.env.NODE_ENV !== 'test') app.use(httpLogger);
 
 // Block requests until service is ready
-app.use((req: Request, res: Response, next: NextFunction): void => {
-  if (state.shutdown) {
-    new Problem(503, { detail: 'Server is shutting down' }).send(req, res);
-  } else if (!state.ready) {
-    new Problem(503, { detail: 'Server is not ready' }).send(req, res);
-  } else {
-    next();
+app.use(async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  if (state.shutdown) return new Problem(503, { detail: 'Server is shutting down' }).send(req, res);
```

Comment:
Would it benefit us to have another state/cache or something that holds the db health/last health check or something that could be checked to avoid extra db hits when the db is down/slow and many requests are coming in?